### PR TITLE
Make Hugging/Petting Interactions Affect Mood

### PR DESCRIPTION
**Pull request by Mnemotechnician**
# Description
Something that was omitted in #733. Hugging and petting now give positive moodlets.

The petting interaction was split into two: one for animals and one for humanoids. The one for animals improves both your own mood and the mood of the animal, whereas petting a humanoid only improves their own mood. In addition to all that, being hugged and being pet do not stack.

<details><summary><h1>Media</h1></summary>
<p>

https://github.com/user-attachments/assets/47e8f722-44ee-4d03-a580-65a2946a1920

</p>
</details>

# Changelog
:cl:
- tweak: Hugging and petting interactions now influence mood, just like the old hugging.

### CODE REVIEW

File: Resources/Locale/en-US/mood/mood.ftl [hunk 1]
```diff
@@ -37,6 +37,8 @@ mood-effect-Dead = You are dead.
 
 mood-effect-BeingHugged = Hugs are nice.
 
+mood-effect-BeingPet = Someone petted me!
```
Comment (DEATHB4DEFEAT) [inline Resources/Locale/en-US/mood/mood.ftl:40]:
```suggestion
mood-effect-BeingPet = Someone pet me!
```

File: Resources/Locale/en-US/mood/mood.ftl [hunk 2]
```diff
@@ -51,4 +53,4 @@ mood-effect-RevolutionFocused = VIVA LA REVOLUTION!!!
 
 mood-effect-CultFocused = Dark Gods, grant me strength!
 
-mood-effect-TraitSanguine = I have nothing to worry about. I'm sure everything will turn out well in the end!
\ No newline at end of file
+mood-effect-TraitSanguine = I have nothing to worry about. I'm sure everything will turn out well in the end!
```
Comment (DEATHB4DEFEAT) [inline Resources/Locale/en-US/mood/mood.ftl:56]:
```suggestion
mood-effect-TraitSanguine = I have nothing to worry about, I'm sure everything will turn out well in the end.
```

File: Resources/Locale/en-US/interaction/verbs/noop.ftl [hunk 3]
```diff
@@ -16,6 +16,12 @@ interaction-Pet-success-self-popup = You pet {THE($target)} on {POSS-ADJ($target
 interaction-Pet-success-target-popup = {THE($user)} pets you on {POSS-ADJ($target)} head.
 interaction-Pet-success-others-popup = {THE($user)} pets {THE($target)}.
 
+interaction-PetAnimal-name = {interaction-Pet-name}
```
Comment (DEATHB4DEFEAT) [inline Resources/Locale/en-US/interaction/verbs/noop.ftl:19]: Is this localization strings in localized strings?

**Review by DEATHB4DEFEAT — CHANGES_REQUESTED** [review]

File: Resources/Locale/en-US/interaction/verbs/noop.ftl (hunk 3 above)
Comment (Mnemotechnician) [reply Resources/Locale/en-US/interaction/verbs/noop.ftl:19]: yes, fluent supports referencing other locale strings from other locale strings

Comment (DEATHB4DEFEAT) [reply Resources/Locale/en-US/interaction/verbs/noop.ftl:19]: Very cool

File: Resources/Locale/en-US/mood/mood.ftl (hunk 2 above)
Comment (Mnemotechnician) [reply Resources/Locale/en-US/mood/mood.ftl:56]: that is not within the scope of this PR.

**Review by DEATHB4DEFEAT — APPROVED** [review]

**Review by VMSolidus — APPROVED** [review]

**Pull request merged**